src/goose/cli/session.py

```diff
@@ -150,23 +154,37 @@ def run(self) -> None:
         """
         message = self.process_first_message()
         while message:  # Loop until no input (empty string).
-            self.notifier.start()
-            try:
-                self.exchange.add(message)
-                self.reply()  # Process the user message.
-            except KeyboardInterrupt:
-                self.interrupt_reply()
-            except Exception:
-                # rewind to right before the last user message
-                self.exchange.rewind()
-                print(traceback.format_exc())
-                print(
-                    "\n[red]The error above was an exception we were not able to handle.\n\n[/]"
-                    + "These errors are often related to connection or authentication\n"
-                    + "We've removed the conversation up to the most recent user message"
-                    + " - [yellow]depending on the error you may be able to continue[/]"
-                )
-            self.notifier.stop()
+            # Start a span with the default tracer. This is present with opentelemetry-instrument
+            with self.tracer.start_as_current_span(message.role) as span:
+                span.set_attribute("goose.role", message.role)
+                span.set_attribute("goose.id", message.id)
+                # For starters, only add to the trace the first text content
+                first_content = message.content[0] if message.content else None
+                if isinstance(first_content, Text):
+                    span.set_attribute("goose.text", first_content.text)
+
+                self.notifier.start()
+                try:
+                    self.exchange.add(message)
+                    self.reply()  # Process the user message.
+                except KeyboardInterrupt:
+                    span.set_status(OtelStatus(OtelStatusCode.ERROR, "KeyboardInterrupt"))
```

Comment:
This is an interesting case. I don't know if an explicit interrupt would actually be an error for the user, it's expected to fail - a browser comparison would be closing a tab during a request, it would be common for the request to still complete on the server side and be green. Anyways, it's probably correct to set it as an error though not fully confident

Reply:
added a comment as I don't know yet, either, but maybe after some practice someone will.

Reply:
hey idea.. maybe we set it to ok, but leave the interrupt?